### PR TITLE
Remove last usages of rusoto in the driver

### PR DESCRIPTION
**Pull request by marcbowes**
Before this commit, there was no actual way to use the awssdk with the
driver (other than how the shell was using it). This commit moves the
abstraction the shell was using into this crate and moves away from
using the rusoto credential providers in the example. Now, we can run
the examples using only the awssdk and not use rusoto at all!

Ideally, we should have -core not generate the client such that -core
remains sdk agnostic. I'll fix that in a later commit.

The awssdk was update to alpha17.

### CODE REVIEW

File: amazon-qldb-driver/Cargo.toml [hunk 1]
```diff
@@ -8,15 +8,13 @@ edition = "2018"
...
 
 [dev-dependencies]
+aws-config = { git = "https://github.com/awslabs/aws-sdk-rust", tag = "v0.0.19-alpha", package = "aws-config", features = [] }
 thiserror = "1.0.29"
-anyhow = "1.0.44"
-tracing = "0.1.28"
-tracing-subscriber = { version = "0.2.21", features = ["fmt"] }
+anyhow = "1.0.43"
+tracing = "0.1.26"
+tracing-subscriber = { version = "0.2.20", features = ["fmt"] }
```
[the remark this answers is not in the document]
Comment (marcbowes) [reply amazon-qldb-driver/Cargo.toml]: Fixed, thanks

File: amazon-qldb-driver/examples/unique.rs [hunk 2]
```diff
@@ -60,11 +62,11 @@ async fn main() -> Result<()> {
     // ```sh
     // qldb> delete from example
     // ```
+    let aws_config = aws_config::load_from_env().await;
+
     let driver = QldbDriverBuilder::new()
         .ledger_name("unique-example")
-        .via_rusoto()
-        .region(Region::UsWest2)
```
Comment (alpian) [inline amazon-qldb-driver/examples/unique.rs:66]: Is this default moved into the config?

**Review by alpian — APPROVED** [review]

File: amazon-qldb-driver/examples/unique.rs (hunk 2 above)
Comment (marcbowes) [reply amazon-qldb-driver/examples/unique.rs:66]: The regional default made no sense. The other defaults are in the config.

File: amazon-qldb-driver/Cargo.toml [hunk 3]
```diff
@@ -8,12 +8,10 @@ edition = "2018"
 
 [dependencies]
 amazon-qldb-driver-core = { version = "*", path = "../amazon-qldb-driver-core" }
-amazon-qldb-driver-rusoto = { version = "*", path = "../amazon-qldb-driver-rusoto" }
-tokio = "1.12.0"
-rusoto_core = { version = "0.47.0", default_features = false, features = ["rustls"] }
-rusoto_qldb_session = { version = "0.47.0", default_features = false, features = ["rustls"] }
+tokio = "1.11.0"
```
Comment (guyilin-amazon) [inline amazon-qldb-driver/Cargo.toml:11]: `tokio` could be `1.12.0` instead? 
https://github.com/awslabs/amazon-qldb-driver-rust/commit/aa00db3640eab45364c199d93e4f25eb0da91c6a#diff-99ebf4a33a9b6967d3f845096b09b1f0b991150a400bd411cb80126f6a1156f6

**Review by guyilin-amazon — APPROVED** [review]
One minor dependency comment

File: amazon-qldb-driver/Cargo.toml (hunk 3 above)
Comment (marcbowes) [reply amazon-qldb-driver/Cargo.toml]: Fixed!

**Pull request merged**